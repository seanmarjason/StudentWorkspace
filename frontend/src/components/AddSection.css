
.addSectionContainer {
  width: 80vw;
  padding: 30px;
  text-align: center;
  border: 2.5px dashed lightgray;
  border-radius: 10px;
  margin: 0 auto;
  cursor: pointer;

}

.addSectionContainer:hover {
  background: lightgray;
  border: 2.5px solid lightgray;
}

.createSection * {
  margin: 10px;
  padding: 5px 10px;
}
