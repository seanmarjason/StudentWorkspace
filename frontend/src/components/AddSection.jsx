import { useState } from 'react';
import axios from 'axios';
import './AddSection.css';

const AddSection = ({ groupId, setWorkspaceData }) => {

  const [createSection, setCreateSection] = useState(false);
  const [newSectionName, setNewSectionName] = useState('');

  const handleCancelClick = (event) => {
    event.preventDefault();
    event.stopPropagation();
    setCreateSection(false);
  }

  const createNewSection = (event) => {
    event.preventDefault();
    if (!newSectionName) {
      return false
    }

    const url = `http://localhost:3000/workspaces/${groupId.toString()}/${newSectionName}`
    
    axios.post(url)
      .then(response => {
        setWorkspaceData(response.data)
        setNewSectionName('');
        setCreateSection(false)
      })
      .catch(error => console.log(`Error: ${error}`))
  }
  
  return (
    <div 
      className="addSectionContainer"
      onClick={() => setCreateSection(true)}
    >
      {
        createSection ?
        <>
          <form
            className="createSection"
            onSubmit={(e) => createNewSection(e)}
          >
            <input 
              id="newSectionNameInput"
              type="text" 
              placeholder='New Section Name'
              value={newSectionName}
              onChange={(e) => setNewSectionName(e.target.value)}
            />
            <button type="submit">Create</button>
            <button onClick={(e) => handleCancelClick(e)}>Cancel</button>
          </form>
        </>
        :
        <p
          className="addSection"
        >
          &#8853; Add New Section
        </p>
      }
    </div>
  )
}


export { AddSection }
