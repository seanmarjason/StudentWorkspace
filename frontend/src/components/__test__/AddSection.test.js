import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import { AddSection } from '../AddSection';

describe('AddSection' , () => {
  it('should match snapshot', async () => {
    const view = render(<AddSection />)

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when user clicks', async () => {
    const view = render(<AddSection />)

    fireEvent(
      screen.getByText('âŠ• Add New Section'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    )

    await waitFor(() => {
      screen.getByText('Create')
    });
    

    expect(view.asFragment()).toMatchSnapshot();
  });
  
})
  