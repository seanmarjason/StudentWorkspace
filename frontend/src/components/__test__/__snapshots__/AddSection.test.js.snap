// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSection should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="addSectionContainer"
  >
    <p
      class="addSection"
    >
      âŠ• Add New Section
    </p>
  </div>
</DocumentFragment>
`;

exports[`AddSection should match snapshot when user clicks 1`] = `
<DocumentFragment>
  <div
    class="addSectionContainer"
  >
    <form
      class="createSection"
    >
      <input
        id="newSectionNameInput"
        placeholder="New Section Name"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Create
      </button>
      <button>
        Cancel
      </button>
    </form>
  </div>
</DocumentFragment>
`;
